lib/java/ydoc-server/src/test/java/org/enso/ydoc/polyfill/web/ZlibTest.java

```diff
@@ -0,0 +1,179 @@
+package org.enso.ydoc.polyfill.web;
+
+import java.util.concurrent.CompletableFuture;
+import org.enso.ydoc.polyfill.ExecutorSetup;
+import org.graalvm.polyglot.Context;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ZlibTest extends ExecutorSetup {
+
+  private static final String TEXT = "Hello World!";
+  private static final String TEXT_BASE64 = "SGVsbG8gV29ybGQh";
+  private static final String TEXT_DEFLATED = "eJzzSM3JyVcIzy/KSVEEABxJBD4=";
+
+  private Context context;
+
+  public ZlibTest() {}
+
+  @Before
+  public void setup() throws Exception {
+    super.setup();
+    var zlib = new Zlib();
+    var contextBuilder = WebEnvironment.createContext();
+
+    context =
+        CompletableFuture.supplyAsync(
+                () -> {
+                  var ctx = contextBuilder.build();
+                  zlib.initialize(ctx);
+                  return ctx;
+                },
+                executor)
+            .get();
+  }
+
+  @After
+  public void tearDown() throws InterruptedException {
+    super.tearDown();
+    context.close();
+  }
+
+  @Test
+  public void bufferFrom() throws Exception {
+    var code = "Buffer.from(TEXT).toString()";
+
+    context.getBindings("js").putMember("TEXT", TEXT);
+
+    var result = CompletableFuture.supplyAsync(() -> context.eval("js", code), executor).get();
```

Comment:
Rather than playing with global variables you should define a JavaScript function:
```java
var code = "(function(text) { return Buffer.from(text).toString(); })";
```
and then invoke it:
```java
var fn = context.eval("js", code);
return fn.execute(TEXT);
```

Reply:
GraalJS sets up tests this way https://github.com/search?q=repo%3Aoracle%2Fgraaljs%20putMember&type=code.  I think it's cleaner than writing a function.